src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -71,4 +89,95 @@ public enum JsonTokenType : byte
+        public void WriteNumber(System.ReadOnlySpan<byte> propertyName, int value, bool suppressEscaping = false) { }
+        public void WriteNumber(System.ReadOnlySpan<byte> propertyName, long value, bool suppressEscaping = false) { }
+        public void WriteNumber(System.ReadOnlySpan<byte> propertyName, float value, bool suppressEscaping = false) { }
+        [System.CLSCompliantAttribute(false)]
+        public void WriteNumber(System.ReadOnlySpan<byte> propertyName, uint value, bool suppressEscaping = false) { }
+        [System.CLSCompliantAttribute(false)]
+        public void WriteNumber(System.ReadOnlySpan<byte> propertyName, ulong value, bool suppressEscaping = false) { }
+        public void WriteNumber(System.ReadOnlySpan<char> propertyName, decimal value, bool suppressEscaping = false) { }
+        public void WriteNumber(System.ReadOnlySpan<char> propertyName, double value, bool suppressEscaping = false) { }
+        public void WriteNumber(System.ReadOnlySpan<char> propertyName, int value, bool suppressEscaping = false) { }
+        public void WriteNumber(System.ReadOnlySpan<char> propertyName, long value, bool suppressEscaping = false) { }
+        public void WriteNumber(System.ReadOnlySpan<char> propertyName, float value, bool suppressEscaping = false) { }
+        [System.CLSCompliantAttribute(false)]
+        public void WriteNumber(System.ReadOnlySpan<char> propertyName, uint value, bool suppressEscaping = false) { }
+        [System.CLSCompliantAttribute(false)]
+        public void WriteNumber(System.ReadOnlySpan<char> propertyName, ulong value, bool suppressEscaping = false) { }
+        public void WriteNumber(string propertyName, decimal value, bool suppressEscaping = false) { }
+        public void WriteNumber(string propertyName, double value, bool suppressEscaping = false) { }
+        public void WriteNumber(string propertyName, int value, bool suppressEscaping = false) { }
+        public void WriteNumber(string propertyName, long value, bool suppressEscaping = false) { }
+        public void WriteNumber(string propertyName, float value, bool suppressEscaping = false) { }
+        [System.CLSCompliantAttribute(false)]
+        public void WriteNumber(string propertyName, uint value, bool suppressEscaping = false) { }
+        [System.CLSCompliantAttribute(false)]
+        public void WriteNumber(string propertyName, ulong value, bool suppressEscaping = false) { }
+        public void WriteNumberValue(decimal value) { }
+        public void WriteNumberValue(double value) { }
+        public void WriteNumberValue(int value) { }
+        public void WriteNumberValue(long value) { }
+        public void WriteNumberValue(float value) { }
+        [System.CLSCompliantAttribute(false)]
+        public void WriteNumberValue(uint value) { }
+        [System.CLSCompliantAttribute(false)]
+        public void WriteNumberValue(ulong value) { }
```

Comment:
We have `WriteNumber` that combines writing the property name with the value, and we have `WriteNumberValue` that just writes the value... is there nothing to just write the property name?

Reply:
I only provided APIs for writing properties with a specific JSON token to follow (like object, number, etc.), mainly as an optimization to collapse as much work into a single call. It also ends up being more convenient for the common case (since the caller usually wants to write `property: value`). We need the WriteValue since the user could be writing elements within a JSON array. **Is there any scenario where the user wants to just write a property that isn't followed by writing a value anyway?**

Btw, not having WritePropery APIs has additional benefits around validation since the token type will never be a property so we can skip certain checks/cases.